src/gt4py/next/program_processors/runners/dace_fieldview/gtir_builtin_translators.py

```diff
@@ -0,0 +1,385 @@
+
+        This method is used by derived classes to build a specialized subgraph
+        for a specific GTIR primitive function.
+
+        Arguments:
+            node: The GTIR node describing the primitive to be lowered
+            sdfg: The SDFG where the primitive subgraph should be instantiated
+            state: The SDFG state where the result of the primitive function should be made available
+            sdfg_builder: The object responsible for visiting child nodes of the primitive node.
+            reduce_identity: The value of the reduction identity, in case the primitive node
+                is visited in the context of a reduction expression. This value is used
+                by the `neighbors` primitive to provide the value of skip neighbors.
+
+        Returns:
+            A list of data access nodes and the associated GT4Py data type, which provide
+            access to the result of the primitive subgraph. The GT4Py data type is useful
+            in the case the returned data is an array, because the type provdes the domain
+            information (e.g. order of dimensions, dimension types).
+        """
+
+
+class AsFieldOp(PrimitiveTranslator):
+    """Generates the dataflow subgraph for the `as_field_op` builtin function."""
+
+    @classmethod
+    def _parse_node_args(
+        cls,
+        args: list[gtir.Expr],
+        sdfg: dace.SDFG,
+        state: dace.SDFGState,
+        sdfg_builder: SDFGBuilder,
+        domain: list[
+            tuple[gtx_common.Dimension, dace.symbolic.SymbolicType, dace.symbolic.SymbolicType]
+        ],
+    ) -> list[gtir_to_tasklet.IteratorExpr | gtir_to_tasklet.MemletExpr]:
+        stencil_args: list[gtir_to_tasklet.IteratorExpr | gtir_to_tasklet.MemletExpr] = []
+        for arg in args:
+            fields: list[TemporaryData] = sdfg_builder.visit(arg, sdfg=sdfg, head_state=state)
+            assert len(fields) == 1
+            data_node, arg_type = fields[0]
+            # require all argument nodes to be data access nodes (no symbols)
+            assert isinstance(data_node, dace.nodes.AccessNode)
+
+            arg_definition: gtir_to_tasklet.IteratorExpr | gtir_to_tasklet.MemletExpr
+            if isinstance(arg_type, ts.ScalarType):
+                arg_definition = gtir_to_tasklet.MemletExpr(data_node, sbs.Indices([0]))
+            else:
+                assert isinstance(arg_type, ts.FieldType)
+                indices: dict[gtx_common.Dimension, gtir_to_tasklet.IteratorIndexExpr] = {
+                    dim: gtir_to_tasklet.SymbolExpr(
+                        dace_fieldview_util.get_map_variable(dim),
+                        IteratorIndexDType,
+                    )
+                    for dim, _, _ in domain
+                }
+                arg_definition = gtir_to_tasklet.IteratorExpr(
+                    data_node,
+                    arg_type.dims,
+                    indices,
+                )
+            stencil_args.append(arg_definition)
```

Comment:
I think this helper function could just deal with the parsing/transformation of one single argument, and then use it in a list comprehension from outside;
```python
stencil_args = [parse_arg_expr(arg) for arg in node.args]
``` 

```suggestion
def _parse_arg_expr(arg: itir.Expr, ....):
    fields: list[TemporaryData] = sdfg_builder.visit(arg, sdfg=sdfg, head_state=state)
    assert len(fields) == 1
    data_node, arg_type = fields[0]
    # require all argument nodes to be data access nodes (no symbols)
    assert isinstance(data_node, dace.nodes.AccessNode)

    arg_definition: gtir_to_tasklet.IteratorExpr | gtir_to_tasklet.MemletExpr
    if isinstance(arg_type, ts.ScalarType):
        arg_definition = gtir_to_tasklet.MemletExpr(data_node, sbs.Indices([0]))
    else:
        assert isinstance(arg_type, ts.FieldType)
        indices: dict[gtx_common.Dimension, gtir_to_tasklet.IteratorIndexExpr] = {
            dim: gtir_to_tasklet.SymbolExpr(
                dace_fieldview_util.get_map_variable(dim),
                IteratorIndexDType,
            )
            for dim, _, _ in domain
        }
        arg_definition = gtir_to_tasklet.IteratorExpr(
            data_node,
            arg_type.dims,
            indices,
        )
    return arg_definition
```

Reply:
Agreed, much better.